File: LargestProductInAGrid/LPG.py
#!/usr/bin/env python3

import re

text = open('pe11.txt')
textContent = text.readlines()
numRegex = re.compile(r'[0-9]+')

g = []
for i in textContent:
    mo = numRegex.findall(i)
    g.append(mo)

Max = 0

for i in range(0,20):
    for j in range(0, 17): # left/right
         l = int(g[i][j])*int(g[i][j+1])*int(g[i][j+2])*int(g[i][j+3])
         if l > Max:
             Max = l
    for j in range(0, 20): # up/down
        if i < 17:
             d =  int(g[i][j])*int(g[i+1][j])*int(g[i+2][j])*int(g[i+3][j])
             if d > Max:
                 Max = d
    for j in range(3, 17): # diagonals
        if i > 2 and i < 17:
            nw = int(g[i][j])*int(g[i-1][j-1])*int(g[i-2][j-2])*int(g[i-3][j-3])
            ne = int(g[i][j])*int(g[i-1][j+1])*int(g[i-2][j+2])*int(g[i-3][j+3])
            sw = int(g[i][j])*int(g[i+1][j-1])*int(g[i+2][j-2])*int(g[i+3][j-3])
            se = int(g[i][j])*int(g[i+1][j+1])*int(g[i+2][j+2])*int(g[i+3][j+3])
            di = max(nw, ne, sw, se)
            if di > Max:
                Max = di
print(Max)

File: LargestProductInAGrid/LPG.java
/*
Project Euler # 11

Java max can only compare two numbers.
*/

import java.io.File;
import java.util.Scanner;

public class LPG
{
    public static void main(String[] args) throws Exception
    {
        Scanner fileIn = new Scanner( new File("pe11.txt") );
        int[][] n = new int[20][20];
        int max = 1;

        for ( int i=0; i<20; i++ )
            for ( int j=0; j<20; j++ )
                n[i][j] = fileIn.nextInt();
        fileIn.close();

        for ( int i=0; i<20; i++ )
        {
            for ( int j=0; j<17; j++ ) // left/right
            {
                int c = n[i][j]*n[i][j+1]*n[i][j+2]*n[i][j+3];
                if ( c > max )
                    max = c;
            }

            for ( int j=0; j<20; j++ )  // up/down
                if ( i < 17 )
                {
                    int c = n[i][j]*n[i+1][j]*n[i+2][j]*n[i+3][j];
                    if ( c > max )
                        max = c;
                }

            for ( int j=3; j<17; j++ )  // diagonals
                if ( i > 2 && i < 17 )
                {
                    int nw = n[i][j]*n[i-1][j-1]*n[i-2][j-2]*n[i-3][j-3];
                    int ne = n[i][j]*n[i-1][j+1]*n[i-2][j+2]*n[i-3][j+3];
                    int sw = n[i][j]*n[i+1][j-1]*n[i+2][j-2]*n[i+3][j-3];
                    int se = n[i][j]*n[i+1][j+1]*n[i+2][j+2]*n[i+3][j+3];
                    int di = Math.max(se,Math.max(sw,Math.max(nw,ne)));
                    if ( di > max )
                      max = di;
                }
        }
        System.out.println(max);

    }
}
